account/identity.go

```diff
@@ -163,6 +164,42 @@ func (m *GormIdentityRepository) Create(ctx context.Context, model *Identity) er
 	return nil
 }
 
+// Lookup looks for an existing identity with the given `profileURL` or creates a new one
+func (m *GormIdentityRepository) Lookup(ctx context.Context, username, profileURL, providerType string) (*Identity, error) {
+	log.Debug(nil, map[string]interface{}{
```

Comment:
Let's check that username and profileURL is not "" here so we are not creating empty identities.